app/importers/mahonia_mapper.rb

```diff
@@ -1,41 +1,40 @@
 # frozen_string_literal: true
 class MahoniaMapper < Darlingtonia::HashMapper
   BEPRESS_TERMS_MAP = {
-    publication_date: :date,
-    title: :title,
-    keywords: :keyword,
-    subject: :subject,
-    document_type: :resource_type,
-    degree_name: :degree,
-    institution_name: :institution,
-    school: :school,
-    department: :department,
-    identifier: :identifier,
-    orcid: :orcid_id,
-    language: :language,
-    license: :license,
-    rights: :rights_statement,
-    note: :rights_note,
-    abstract: :description,
-    journal_title: :source
+    date: "publication_date",
+    title: "title",
+    keyword: "keywords",
+    subject: "subject",
+    resource_type: "document_type",
+    degree: "degree_name",
+    institution: "institution_name",
+    school: "school",
+    department: "department",
+    identifier: "identifier",
+    orcid_id: "orcid",
+    language: "language",
+    license: "license",
+    rights_statement: "rights",
+    rights_note: "note",
+    description: "abstract",
+    source: "journal_title",
+    creator: "creator"
   }.freeze
 
-  def metadata=(meta)
-    be_press_hash = meta.to_h
-    @metadata = {}
-    be_press_hash.each_key do |k|
-      if BEPRESS_TERMS_MAP.keys.include? k.to_sym
-        @metadata[BEPRESS_TERMS_MAP[k.to_sym]] = be_press_hash.delete k
-      end
+  def fields
+    BEPRESS_TERMS_MAP.keys.reject do |e|
```

Comment:
Do we need to reject `:creator`? I think this will keep it from getting into the `attributes` hash?

Reply:
The creator method determines whether it has any data to return, and it seemed a little cleaner than trying to find that out now.

Reply:
oh, i see, we are *never* rejecting `:creator`. that seems :+1:, but I think it complicates your argument in https://github.com/curationexperts/mahonia/pull/251/files#r162716045.

The main advantage of a static `#fields`, besides a general reduced dependence on object state, seems to be that it avoids any need for special cases. `#fields` always returns the method calls a consumer will want to know about, and methods always return the relevant values.

Reply:
I guess then it might be best to write a method for each field that returns if there's no data? This just seems like a nice filter to me, fields only returning methods that return data.

Reply:
I'm okay with either way. I was confused for a moment by the handling of `:creator` but I see that this works now and am happy to merge it in.

Reply:
Okay, thanks. I'm glad it seems okay to you.